core/src/main/java/org/apache/iceberg/ManifestFilterManager.java

```diff
@@ -78,9 +78,11 @@ public String partition() {
   private boolean failMissingDeletePaths = false;
   private int duplicateDeleteCount = 0;
   private boolean caseSensitive = true;
+  private boolean allDeletesReferenceManifests = true;
```

Comment:
The downside of defaulting true and setting false is that every place which exposes a deletion where it's impossible to reference a manifest needs to be updated. But that seemed like only a handful of places, and probably won't grow?

Worth considering other approaches where this defaults false, and at the end we figure out if it's true. I think this would require keeping track of a map<string, set<String>> manifestToFiles and comparing the set with the deletePaths...but that implies duplicating all the paths in memory.